packages/native/src/index.js

```diff
@@ -0,0 +1,71 @@
+import reactNative from 'react-native'
+import { createCss } from '@emotion/primitives-core'
+
+import { styled } from './styled'
+
+const css = createCss(reactNative.StyleSheet)
+
+const components = [
+  'ActivityIndicator',
+  'ActivityIndicatorIOS',
+  'ART',
+  'Button',
+  'DatePickerIOS',
+  'DrawerLayoutAndroid',
+  'Image',
+  'ImageBackground',
+  'ImageEditor',
+  'ImageStore',
+  'KeyboardAvoidingView',
+  'ListView',
+  'MapView',
+  'Modal',
+  'NavigatorIOS',
+  'Picker',
+  'PickerIOS',
+  'ProgressBarAndroid',
+  'ProgressViewIOS',
+  'ScrollView',
+  'SegmentedControlIOS',
+  'Slider',
+  'SliderIOS',
+  'SnapshotViewIOS',
+  'Switch',
+  'RecyclerViewBackedScrollView',
+  'RefreshControl',
+  'SafeAreaView',
+  'StatusBar',
+  'SwipeableListView',
+  'SwitchAndroid',
+  'SwitchIOS',
+  'TabBarIOS',
+  'Text',
+  'TextInput',
+  'ToastAndroid',
+  'ToolbarAndroid',
+  'Touchable',
+  'TouchableHighlight',
+  'TouchableNativeFeedback',
+  'TouchableOpacity',
+  'TouchableWithoutFeedback',
+  'View',
+  'ViewPagerAndroid',
+  'WebView',
+  'FlatList',
+  'SectionList',
+  'VirtualizedList'
+]
+
+components.forEach(comp =>
+  Object.defineProperty(styled, comp, {
+    enumerable: true,
+    configurable: false,
+    get() {
+      return styled(reactNative[comp])
+    }
+  })
+)
+
+export { css }
+
+export default /* #__PURE__ */ styled
```

Comment:
this doesnt make much sense here, `#__PURE__` comments are useful for call expressions, this is just exported identifier

Not sure what actual value it gives in context of react-native (it's not that much bytes-constrained as web), it certainly won't hurt, but would have to be used differently.

I would convert this to:
```js
export default components.reduce((acc, comp) =>
  Object.defineProperty(acc, comp, {
    enumerable: true,
    configurable: false,
    get() {
      return styled(reactNative[comp])
    }
  })
, styled)
```

This would make it self contained expression that can be more safely annotated with PURE.

You could also use my plugin to insert those PURE annotations automatically - https://github.com/Andarist/babel-plugin-annotate-pure-calls , but ofc that's not obligatory 😉

Reply:
That makes sense. I'll refactor the code to use your solution.